modules/battery_manager/ajax/add_entry.php

```diff
@@ -0,0 +1,137 @@
+<?php
+/**
+ * Battery Manager adder
+ *
+ * Checks Test Battery for duplicates and handles insertions into Test Battery
+ *
+ * PHP Version 7
+ *
+ * @category Loris
+ * @package  Battery_Manager
+ * @author   Victoria Foing <victoria.foing@mcin.ca>
+ * @license  Loris license
+ * @link     https://github.com/aces/Loris-Trunk
+ */
+
+
+// Determine which action is called
+if (isset($_GET['action'])) {
+    $action = $_GET['action'];
+    if ($action == "checkForDuplicate") {
+        echo checkForDuplicate();
+    } else if ($action == "add") {
+        addEntry();
+    }
+}
+
+/**
```

Comment:
this check seems to be done on the front-end which is okai but should also happen on the back end since the front end JS code can be manipulated by the user !

validation in general is minimally on the backend, max front end and back end but never FE alone